MockKit/MockDoseProgressEstimator.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  MockDoseProgressEstimator.swift
+//  MockKit
+//
+//  Created by Pete Schwamb on 3/17/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import LoopKit
+
+class MockDoseProgressEstimator: DoseProgressEstimator {
+
+
+    public let dose: DoseEntry
+
+    private var observers = WeakSet<DoseProgressObserver>()
+
+    private var timer: Timer?
+
+    private var lock = os_unfair_lock()
+
+    var progress: DoseProgress {
+        os_unfair_lock_lock(&lock)
+        defer {
+            os_unfair_lock_unlock(&lock)
+        }
```

Comment:
[An article I came across recently](http://www.russbishop.net/the-law) describes how Swift's exclusivity model makes the expected use of os_unfair_lock unsafe. We should favor NSLock or DispatchQueue.

Reply:
Added new UnfairLock, and updated existing Lock to deal with potential exclusivity enforcement.